docs/fields/select.mdx

```diff
@@ -67,6 +68,61 @@ _\* An asterisk denotes that a property is required._
   used as a GraphQL enum.
 </Banner>
 
+### reduceOptions
+
+Used to dynamically filter which options are available based on the user, data, etc.
+
+Some examples of this might include:
+
+- Restricting options based on a user's role, e.g. admin-only options
+- Displaying different options based on the value of another field, e.g. a city/state selector
+
+The result of `reduceOptions` will determine:
+
+- Which options are displayed in the Admin Panel
+- Which options can be saved to the database
+
+To do this, use the `reduceOptions` property in your [Field Config](./overview):
+
+```ts
+import type { Field } from 'payload'
+
+export const MySelectField: Field = {
+  // ...
+  // highlight-start
+  type: 'select',
+  options: [
+    {
+      label: 'One',
+      value: 'one',
+    },
+    {
+      label: 'Two',
+      value: 'two',
+    },
+    {
+      label: 'Three',
+      value: 'three',
+    },
+  ],
+  reduceOptions: ({ options, data }) =>
```

Comment:
I am on the fence about this being called something different than `filterOptions`. I get that the type is going to be different than `relationship` and `upload` fields but the functionality is the same. Internal to Payload this might be slightly confusing for us, but we can add our own type as `SelectFilterOptions` and leave the relationship and uploads one alone. Since react-select calls this `filterOption` too it makes sense to keep it standard might which will be less for devs to learn and remember.

Reply:
I 100% agree with this. Done in [ad6311b](https://github.com/payloadcms/payload/pull/12487/commits/ad6311bce6494349fbb5fa3897f37cd5369cf526).